ulauncher/search/apps/AppResultItem.py

```diff
@@ -1,42 +1,87 @@
+from os.path import basename
+import gi
+gi.require_version('Gio', '2.0')
+# pylint: disable=wrong-import-position
+from gi.repository import Gio
+from ulauncher.utils.Settings import Settings
+from ulauncher.utils.fuzzy_search import get_score
 from ulauncher.api.shared.action.LaunchAppAction import LaunchAppAction
 from ulauncher.api.shared.item.ResultItem import ResultItem
 from ulauncher.search.QueryHistoryDb import QueryHistoryDb
 from ulauncher.search.apps.AppStatDb import AppStatDb
+from ulauncher.utils.image_loader import get_app_icon_pixbuf
+
+settings = Settings.get_instance()
+disable_desktop_filters = settings.get_property('disable-desktop-filters')
+blacklisted_dirs = settings.get_property('blacklisted-desktop-dirs')
+blacklisted_dirs_list = blacklisted_dirs.split(':') if blacklisted_dirs else []
 
 
 class AppResultItem(ResultItem):
     """
-    :param dict record:
+    :param Gio.DesktopAppInfo app_info:
     """
-
     # pylint: disable=super-init-not-called
-    def __init__(self, record):
-        self.record = record
+    def __init__(self, app_info):
+        self._name = app_info.get_display_name()
+        # TryExec is what we actually want (name of/path to exec), but it's often not specified
+        # get_executable uses Exec, which is always specified, but it will return the actual executable.
+        # Sometimes the actual executable is not the app to start, but a wrappers like "env" or "sh -c"
+        self._executable = basename(app_info.get_string('TryExec') or app_info.get_executable())
+        self._app_info = app_info
         self._query_history = QueryHistoryDb.get_instance()
         self._app_stat_db = AppStatDb.get_instance()
 
-    def get_name(self):
-        return self.record.get('name')
+    @staticmethod
+    def from_id(app_id):
+        try:
+            app_info = Gio.DesktopAppInfo.new(app_id)
+            return AppResultItem(app_info)
+        except TypeError:
+            return None
+
+    @staticmethod
+    def search(query, min_score=50, limit=9, apps=Gio.DesktopAppInfo.get_all()):
```

Comment:
Using `get_all()` as a default argument means it is called at import time, and then never updated. better to set as `None` and then check in the body of the function.

Reply:
It was inside of the block before but I moved it to a default argument so I would be able to mock it. Was still working on the tests when I ran into other issues.